{
	"id": "166895",
	"key": "TIMOB-24548",
	"fields": {
		"issuetype": {
			"id": "1",
			"description": "A problem which impairs or prevents the functions of the product.",
			"name": "Bug",
			"subtask": false
		},
		"project": {
			"id": "10153",
			"key": "TIMOB",
			"name": "Titanium SDK/CLI",
			"projectCategory": {
				"id": "10100",
				"description": "Titanium and related SDKs used in application development",
				"name": "Client"
			}
		},
		"fixVersions": [
			{
				"id": "11258",
				"description": "Holding Pen for Triaged Issues",
				"name": "Backlog",
				"archived": false,
				"released": false
			}
		],
		"resolution": {
			"id": "11",
			"description": "Is not a bug in our product",
			"name": "Not Our Bug"
		},
		"resolutiondate": "2017-05-18T16:39:23.000+0000",
		"created": "2017-03-31T23:20:07.000+0000",
		"priority": {
			"name": "High",
			"id": "2"
		},
		"labels": [],
		"versions": [],
		"issuelinks": [],
		"assignee": {
			"name": "jquick",
			"key": "jquick",
			"displayName": "Joshua Quick",
			"active": true,
			"timeZone": "America/Los_Angeles"
		},
		"updated": "2017-05-30T16:45:22.000+0000",
		"status": {
			"description": "The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.",
			"name": "Closed",
			"id": "6",
			"statusCategory": {
				"id": 3,
				"key": "done",
				"colorName": "green",
				"name": "Done"
			}
		},
		"components": [
			{
				"id": "10202",
				"name": "Android",
				"description": "Android Platform"
			}
		],
		"description": "While using generated SDK code from API Portal, unable to run the mobile application based on sdk code. After following steps mentioned in readme, We also tried couple of different ways to resolve it as suggested by SDK team, but it seems that alloy project is unable to recognize those files (of sdk code). Due to which we get error for module not found.\r\n\r\nAs of now we have followed each step given in readme but when trying to run it, application returns below error:\r\n\r\n[ERROR] :  TiExceptionHandler: (main) [1074,1074] ----- Titanium Javascript Runtime Error -----\r\n[ERROR] :  TiExceptionHandler: (main) [1,1075] - In ti:/module.js:303,2\r\n[ERROR] :  TiExceptionHandler: (main) [1,1076] - Message: Uncaught Error: Requested module not found: ./studentcar/index\r\n[ERROR] :  TiExceptionHandler: (main) [0,1076] - Source:     throw new Error(\"Requested module not found: \" + request); // TODO Set 'code' property to 'MODULE_NOT_FOUND' to match Node?\r\n[ERROR] :  V8Exception: Exception occurred at ti:/module.js:303: Uncaught Error: Requested module not found: ./studentcar/index\r\n\r\nExpected Result: Using generated SDK code from mobile application user should be able to access APIs generated in Arrow and make respective calls.\r\n\r\n",
		"attachment": [
			{
				"id": "61967",
				"filename": "studentcar_titanium.zip",
				"author": {
					"name": "ingo",
					"key": "ingo",
					"displayName": "Ingo Muschenetz",
					"active": true,
					"timeZone": "America/Los_Angeles"
				},
				"created": "2017-04-05T13:31:08.000+0000",
				"size": 114202,
				"mimeType": "application/zip"
			},
			{
				"id": "61968",
				"filename": "studentcar.json",
				"author": {
					"name": "ingo",
					"key": "ingo",
					"displayName": "Ingo Muschenetz",
					"active": true,
					"timeZone": "America/Los_Angeles"
				},
				"created": "2017-04-05T13:31:06.000+0000",
				"size": 92255,
				"mimeType": "application/json"
			}
		],
		"flagged": false,
		"summary": "Android: Issue with SDK while accessing from API Gateway",
		"creator": {
			"name": "anarwal",
			"key": "anarwal",
			"displayName": "Abhimanyu Narwal",
			"active": true,
			"timeZone": "America/Los_Angeles"
		},
		"subtasks": [],
		"reporter": {
			"name": "anarwal",
			"key": "anarwal",
			"displayName": "Abhimanyu Narwal",
			"active": true,
			"timeZone": "America/Los_Angeles"
		},
		"environment": null,
		"closedSprints": [
			{
				"id": 883,
				"state": "closed",
				"name": "2017 Sprint 09 SDK",
				"startDate": "2017-04-23T21:46:54.309Z",
				"endDate": "2017-05-07T21:46:00.000Z",
				"completeDate": "2017-05-08T00:25:44.276Z",
				"originBoardId": 114
			}
		],
		"comment": {
			"comments": [
				{
					"id": "416818",
					"author": {
						"name": "ingo",
						"key": "ingo",
						"displayName": "Ingo Muschenetz",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"body": "I ran into a similar issue.\r\n\r\n# I updated the API gateway to the latest 1.0.0 version. That gave me the attached SDK.\r\n# I integrated it at a superficial level, just updated alloy.js\r\n# I ran it on iOS. Worked.\r\n# I ran it on Android. Failed with a similar message about \"unable to find ../model/student\"\r\n# I noticed that my files are capitalized (as in \"Student.js\")\r\n# Editing the DefaultAPI.js to reflect the following worked:\r\n\r\n{code}\r\nvar Client = require('../client'),\r\n\tstudent = require('../model/Student'),\r\n\ttestuser = require('../model/Testuser'),\r\n\tcar = require('../model/Car');\r\n{code}\r\n\r\n# My OS X boot volume is NOT formatted case-sensitive\r\n# Android _is_ case sensitive: https://stackoverflow.com/questions/6502712/isnt-android-file-exists-case-sensitive\r\n\r\nMy sense is that https://github.com/appcelerator/client-generator/blob/command_line/templates/titanium/api.js#L8 needs to respect the capitalization.",
					"updateAuthor": {
						"name": "ingo",
						"key": "ingo",
						"displayName": "Ingo Muschenetz",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"created": "2017-04-05T04:55:24.000+0000",
					"updated": "2017-04-05T04:55:49.000+0000"
				},
				{
					"id": "416875",
					"author": {
						"name": "emerriman",
						"key": "emerriman",
						"displayName": "Eric Merriman ",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"body": "Josh, can you take a look at this and let me know what you think?",
					"updateAuthor": {
						"name": "emerriman",
						"key": "emerriman",
						"displayName": "Eric Merriman ",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"created": "2017-04-05T20:40:40.000+0000",
					"updated": "2017-04-05T20:40:40.000+0000"
				},
				{
					"id": "416885",
					"author": {
						"name": "jquick",
						"key": "jquick",
						"displayName": "Joshua Quick",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"body": "The newest \"client-generator\" (v1.0.0-17) and \"ti.oauth\" (v0.3.1) changes will solve this issue.",
					"updateAuthor": {
						"name": "jquick",
						"key": "jquick",
						"displayName": "Joshua Quick",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"created": "2017-04-05T21:41:44.000+0000",
					"updated": "2017-04-05T22:15:37.000+0000"
				},
				{
					"id": "419952",
					"author": {
						"name": "ingo",
						"key": "ingo",
						"displayName": "Ingo Muschenetz",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"body": "This is really a client SDK bug, not a Titanium Mobile one.",
					"updateAuthor": {
						"name": "ingo",
						"key": "ingo",
						"displayName": "Ingo Muschenetz",
						"active": true,
						"timeZone": "America/Los_Angeles"
					},
					"created": "2017-05-18T16:39:23.000+0000",
					"updated": "2017-05-18T16:39:23.000+0000"
				},
				{
					"id": "420609",
					"author": {
						"name": "lmorris",
						"key": "lmorris",
						"displayName": "Lee Morris",
						"active": false,
						"timeZone": "America/Los_Angeles"
					},
					"body": "Closing ticket with reference to the above comments.",
					"updateAuthor": {
						"name": "lmorris",
						"key": "lmorris",
						"displayName": "Lee Morris",
						"active": false,
						"timeZone": "America/Los_Angeles"
					},
					"created": "2017-05-30T16:45:22.000+0000",
					"updated": "2017-05-30T16:45:22.000+0000"
				}
			],
			"maxResults": 5,
			"total": 5,
			"startAt": 0
		}
	}
}